app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,57 @@
+class OrdersController < ApplicationController
+    before_action :order_page_access, only: [:index]
+
+  def index
+    @merchant = session[:user_id]
+    @orders = Order.includes(:order_items).where(order_items: { user_id: @merchant } )
+    # @order_item = OrderItem.find(params[:id])
+
+    if params[:status] == "paid"
+      @orders = @orders.where("status = ?", "paid")
+    elsif params[:status] == "complete"
+      @orders = @orders.where("status = ?", "Complete")
+    end
+    render :index
+  end
+
+
+  def edit
+    @order = Order.find(session[:order_id])
+    render :edit
+  end
+
+  def show
+    @order = Order.find(params[:order_id])
+    render :show
+  end
+
+  def update
+
+    @order = Order.find(session[:order_id])
+    @order.update(create_params[:checkout])
```

Comment:
In tracing through this method, it does at least three total save operations. One here with the `update` one later again on the `@order` and again in the `update_stock` method for each updated product. It would be worthwhile to see if there is a way to reduce at least this first duplication of saves.